### PR TITLE
Integrate the ECS and Physics

### PR DESCRIPTION
**Pull request by foodelevator**
Depends on #37

Integrates the ECS with the physics.

The way it currently works is that `game_engine::physics_systems::update` takes a `&mut ecs::World` and runs a physics step on all entities with a `common::Transform`, `physics::Rigidbody`, and optionally `physics::Collider`. It also looks for a `physics::Gravity` and `game_engine::Time` as a resource.

### CODE REVIEW

**Review by VincentLagerros — COMMENTED** [review]
Looks like a good refactor 👍 

I do wonder if renaming everything from box -> cube or cuboid is better to not fuck with std::box

**Comment by mathiasmagnusson** [comment]
In my opinion having to do `r#box` is pretty ugly, and I guess it could decrease confusion, so I agree 👍

File: physics/src/sphere.rs [hunk 1]
```diff
@@ -19,9 +21,12 @@ impl SphereColider {
     }
 
     pub fn get_radius(&self, scale: Vec3) -> f32 {
-        let scale = scale.x; // TODO FIX, this is left to the user to discover
+        // TODO: add support for non-uniformly scaled "spheres"
```
[the remark this answers is not in the document]
Comment (mathiasmagnusson) [reply physics/src/sphere.rs:24]: Indeed no, but it looks better this way :^)

File: physics/src/collision.rs [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+    sphere::collision::collide_sphere_vs_sphere,
+    sphere::collision::{is_colliding_sphere_vs_box, is_colliding_sphere_vs_sphere},
+    sphere::{collision::collide_sphere_vs_box, SphereCollider},
+    PhysicsMaterial, Rigidbody,
+};
+
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum Collider {
+    Sphere(SphereCollider),
+    Box(BoxCollider),
```
[the remark this answers is not in the document]
Comment (mathiasmagnusson) [reply physics/src/collision.rs:18]: It confuses most syntax highlighters and perhaps some people, but not the compiler

**Review by VincentLagerros — APPROVED** [review]

File: physics/src/sphere.rs [hunk 3]
```diff
@@ -19,9 +21,12 @@ impl SphereColider {
...
+        let scale = (scale.x + scale.y + scale.z) / 3.;
```
Comment (danieltottie) [inline physics/src/sphere.rs:25]: nice xD
I'm a bit out of the loop regarding the physics, but have we thought about using capsule colliders for ellipsoids? intuitively I feel like it should be the cheapest option but idk?

**Review by RealButterbee — APPROVED** [review]
took a while to read through but it looks good!

File: physics/src/sphere.rs (hunk 3 above)
Comment (LagradOst) [reply physics/src/sphere.rs:25]: No, I dont want to add another shape atm

**Pull request merged**